File: src/goopylib/color/W3CX11.cpp
#define GP_LOGGING_LEVEL 3
#include "W3CX11.h"


namespace gp {
    GPAPI std::unordered_map<std::string, Scope<ColorHex>> colors;

    void initW3CX11() {
        GP_CORE_DEBUG("gp::initW3CX11()");

        colors["aliceblue"] = CreateScope<ColorHex>("f0f8ff");
        colors["antiquewhite"] = CreateScope<ColorHex>("faebd7");
        colors["aqua"] = CreateScope<ColorHex>("00ffff");
        colors["aquamarine"] = CreateScope<ColorHex>("7fffd4");
        colors["azure"] = CreateScope<ColorHex>("f0ffff");
        colors["beige"] = CreateScope<ColorHex>("f5f5dc");
        colors["bisque"] = CreateScope<ColorHex>("ffe4c4");
        colors["black"] = CreateScope<ColorHex>("000000");
        colors["blanchedalmond"] = CreateScope<ColorHex>("ffebcd");
        colors["blue"] = CreateScope<ColorHex>("0000ff");
        colors["blueviolet"] = CreateScope<ColorHex>("8a2be2");
        colors["brown"] = CreateScope<ColorHex>("a52a2a");
        colors["burlywood"] = CreateScope<ColorHex>("deb887");
        colors["cadetblue"] = CreateScope<ColorHex>("5f9ea0");
        colors["chartreuse"] = CreateScope<ColorHex>("7fff00");
        colors["chocolate"] = CreateScope<ColorHex>("d2691e");
        colors["coral"] = CreateScope<ColorHex>("ff7f50");
        colors["cornflower"] = CreateScope<ColorHex>("6495ed");
        colors["cornflowerblue"] = CreateScope<ColorHex>("6495ed");
        colors["cornsilk"] = CreateScope<ColorHex>("fff8dc");
        colors["crimson"] = CreateScope<ColorHex>("dc143c");
        colors["cyan"] = CreateScope<ColorHex>("00ffff");
        colors["darkblue"] = CreateScope<ColorHex>("00008b");
        colors["darkcyan"] = CreateScope<ColorHex>("008b8b");
        colors["darkgoldenrod"] = CreateScope<ColorHex>("b8860b");
        colors["darkgray"] = CreateScope<ColorHex>("a9a9a9");
        colors["darkgreen"] = CreateScope<ColorHex>("006400");
        colors["darkgrey"] = CreateScope<ColorHex>("a9a9a9");
        colors["darkkhaki"] = CreateScope<ColorHex>("bdb76b");
        colors["darkmagenta"] = CreateScope<ColorHex>("8b008b");
        colors["darkolivegreen"] = CreateScope<ColorHex>("556b2f");
        colors["darkorange"] = CreateScope<ColorHex>("ff8c00");
        colors["darkorchid"] = CreateScope<ColorHex>("9932cc");
        colors["darkred"] = CreateScope<ColorHex>("8b0000");
        colors["darksalmon"] = CreateScope<ColorHex>("e9967a");
        colors["darkseagreen"] = CreateScope<ColorHex>("8fbc8f");
        colors["darkslateblue"] = CreateScope<ColorHex>("483d8b");
        colors["darkslategray"] = CreateScope<ColorHex>("2f4f4f");
        colors["darkslategrey"] = CreateScope<ColorHex>("2f4f4f");
        colors["darkturquoise"] = CreateScope<ColorHex>("00ced1");
        colors["darkviolet"] = CreateScope<ColorHex>("9400d3");
        colors["deeppink"] = CreateScope<ColorHex>("ff1493");
        colors["deepskyblue"] = CreateScope<ColorHex>("00bfff");
        colors["dimgray"] = CreateScope<ColorHex>("696969");
        colors["dimgrey"] = CreateScope<ColorHex>("696969");
        colors["dodgerblue"] = CreateScope<ColorHex>("1e90ff");
        colors["firebrick"] = CreateScope<ColorHex>("b22222");
        colors["floralwhite"] = CreateScope<ColorHex>("fffaf0");
        colors["forestgreen"] = CreateScope<ColorHex>("228b22");
        colors["fuchsia"] = CreateScope<ColorHex>("ff00ff");
        colors["gainsboro"] = CreateScope<ColorHex>("dcdcdc");
        colors["ghostwhite"] = CreateScope<ColorHex>("f8f8ff");
        colors["gold"] = CreateScope<ColorHex>("ffd700");
        colors["goldenrod"] = CreateScope<ColorHex>("daa520");
        colors["gray"] = CreateScope<ColorHex>("808080");
        colors["green"] = CreateScope<ColorHex>("008000");
        colors["greenyellow"] = CreateScope<ColorHex>("adff2f");
        colors["grey"] = CreateScope<ColorHex>("808080");
        colors["honeydew"] = CreateScope<ColorHex>("f0fff0");
        colors["hotpink"] = CreateScope<ColorHex>("ff69b4");
        colors["indianred"] = CreateScope<ColorHex>("cd5c5c");
        colors["indigo"] = CreateScope<ColorHex>("4b0082");
        colors["ivory"] = CreateScope<ColorHex>("fffff0");
        colors["khaki"] = CreateScope<ColorHex>("f0e68c");
        colors["laserlemon"] = CreateScope<ColorHex>("ffff54");
        colors["lavender"] = CreateScope<ColorHex>("e6e6fa");
        colors["lavenderblush"] = CreateScope<ColorHex>("fff0f5");
        colors["lawngreen"] = CreateScope<ColorHex>("7cfc00");
        colors["lemonchiffon"] = CreateScope<ColorHex>("fffacd");
        colors["lightblue"] = CreateScope<ColorHex>("add8e6");
        colors["lightcoral"] = CreateScope<ColorHex>("f08080");
        colors["lightcyan"] = CreateScope<ColorHex>("e0ffff");
        colors["lightgoldenrod"] = CreateScope<ColorHex>("fafad2");
        colors["lightgoldenrodyellow"] = CreateScope<ColorHex>("fafad2");
        colors["lightgray"] = CreateScope<ColorHex>("d3d3d3");
        colors["lightgreen"] = CreateScope<ColorHex>("90ee90");
        colors["lightgrey"] = CreateScope<ColorHex>("d3d3d3");
        colors["lightpink"] = CreateScope<ColorHex>("ffb6c1");
        colors["lightsalmon"] = CreateScope<ColorHex>("ffa07a");
        colors["lightseagreen"] = CreateScope<ColorHex>("20b2aa");
        colors["lightskyblue"] = CreateScope<ColorHex>("87cefa");
        colors["lightslategray"] = CreateScope<ColorHex>("778899");
        colors["lightslategrey"] = CreateScope<ColorHex>("778899");
        colors["lightsteelblue"] = CreateScope<ColorHex>("b0c4de");
        colors["lightyellow"] = CreateScope<ColorHex>("ffffe0");
        colors["lime"] = CreateScope<ColorHex>("00ff00");
        colors["limegreen"] = CreateScope<ColorHex>("32cd32");
        colors["linen"] = CreateScope<ColorHex>("faf0e6");
        colors["magenta"] = CreateScope<ColorHex>("ff00ff");
        colors["maroon"] = CreateScope<ColorHex>("800000");
        colors["maroon2"] = CreateScope<ColorHex>("7f0000");
        colors["maroon3"] = CreateScope<ColorHex>("b03060");
        colors["mediumaquamarine"] = CreateScope<ColorHex>("66cdaa");
        colors["mediumblue"] = CreateScope<ColorHex>("0000cd");
        colors["mediumorchid"] = CreateScope<ColorHex>("ba55d3");
        colors["mediumpurple"] = CreateScope<ColorHex>("9370db");
        colors["mediumseagreen"] = CreateScope<ColorHex>("3cb371");
        colors["mediumslateblue"] = CreateScope<ColorHex>("7b68ee");
        colors["mediumspringgreen"] = CreateScope<ColorHex>("00fa9a");
        colors["mediumturquoise"] = CreateScope<ColorHex>("48d1cc");
        colors["mediumvioletred"] = CreateScope<ColorHex>("c71585");
        colors["midnightblue"] = CreateScope<ColorHex>("191970");
        colors["mintcream"] = CreateScope<ColorHex>("f5fffa");
        colors["mistyrose"] = CreateScope<ColorHex>("ffe4e1");
        colors["moccasin"] = CreateScope<ColorHex>("ffe4b5");
        colors["navajowhite"] = CreateScope<ColorHex>("ffdead");
        colors["navy"] = CreateScope<ColorHex>("000080");
        colors["oldlace"] = CreateScope<ColorHex>("fdf5e6");
        colors["olive"] = CreateScope<ColorHex>("808000");
        colors["olivedrab"] = CreateScope<ColorHex>("6b8e23");
        colors["orange"] = CreateScope<ColorHex>("ffa500");
        colors["orangered"] = CreateScope<ColorHex>("ff4500");
        colors["orchid"] = CreateScope<ColorHex>("da70d6");
        colors["palegoldenrod"] = CreateScope<ColorHex>("eee8aa");
        colors["palegreen"] = CreateScope<ColorHex>("98fb98");
        colors["paleturquoise"] = CreateScope<ColorHex>("afeeee");
        colors["palevioletred"] = CreateScope<ColorHex>("db7093");
        colors["papayawhip"] = CreateScope<ColorHex>("ffefd5");
        colors["peachpuff"] = CreateScope<ColorHex>("ffdab9");
        colors["peru"] = CreateScope<ColorHex>("cd853f");
        colors["pink"] = CreateScope<ColorHex>("ffc0cb");
        colors["plum"] = CreateScope<ColorHex>("dda0dd");
        colors["powderblue"] = CreateScope<ColorHex>("b0e0e6");
        colors["purple"] = CreateScope<ColorHex>("800080");
        colors["purple2"] = CreateScope<ColorHex>("7f007f");
        colors["purple3"] = CreateScope<ColorHex>("a020f0");
        colors["rebeccapurple"] = CreateScope<ColorHex>("663399");
        colors["red"] = CreateScope<ColorHex>("ff0000");
        colors["rosybrown"] = CreateScope<ColorHex>("bc8f8f");
        colors["royalblue"] = CreateScope<ColorHex>("4169e1");
        colors["saddlebrown"] = CreateScope<ColorHex>("8b4513");
        colors["salmon"] = CreateScope<ColorHex>("fa8072");
        colors["sandybrown"] = CreateScope<ColorHex>("f4a460");
        colors["seagreen"] = CreateScope<ColorHex>("2e8b57");
        colors["seashell"] = CreateScope<ColorHex>("fff5ee");
        colors["sienna"] = CreateScope<ColorHex>("a0522d");
        colors["silver"] = CreateScope<ColorHex>("c0c0c0");
        colors["skyblue"] = CreateScope<ColorHex>("87ceeb");
        colors["slateblue"] = CreateScope<ColorHex>("6a5acd");
        colors["slategray"] = CreateScope<ColorHex>("708090");
        colors["slategrey"] = CreateScope<ColorHex>("708090");
        colors["snow"] = CreateScope<ColorHex>("fffafa");
        colors["springgreen"] = CreateScope<ColorHex>("00ff7f");
        colors["steelblue"] = CreateScope<ColorHex>("4682b4");
        colors["tan"] = CreateScope<ColorHex>("d2b48c");
        colors["teal"] = CreateScope<ColorHex>("008080");
        colors["thistle"] = CreateScope<ColorHex>("d8bfd8");
        colors["tomato"] = CreateScope<ColorHex>("ff6347");
        colors["turquoise"] = CreateScope<ColorHex>("40e0d0");
        colors["violet"] = CreateScope<ColorHex>("ee82ee");
        colors["wheat"] = CreateScope<ColorHex>("f5deb3");
        colors["white"] = CreateScope<ColorHex>("ffffff");
        colors["whitesmoke"] = CreateScope<ColorHex>("f5f5f5");
        colors["yellow"] = CreateScope<ColorHex>("ffff00");
        colors["yellowgreen"] = CreateScope<ColorHex>("9acd32");
    }
}

File: goopylib/color/w3cx11.py
"""
Module defining colors listed by the W3CX11 specification
"""

from __future__ import annotations

from typing import Dict

# pylint: disable-next=no-name-in-module
from . import ColorHex


colors: Dict[str, ColorHex] = {
    "aliceblue":            ColorHex("f0f8ff"),
    "antiquewhite":         ColorHex("faebd7"),
    "aqua":                 ColorHex("00ffff"),
    "aquamarine":           ColorHex("7fffd4"),
    "azure":                ColorHex("f0ffff"),
    "beige":                ColorHex("f5f5dc"),
    "bisque":               ColorHex("ffe4c4"),
    "black":                ColorHex("000000"),
    "blanchedalmond":       ColorHex("ffebcd"),
    "blue":                 ColorHex("0000ff"),
    "blueviolet":           ColorHex("8a2be2"),
    "brown":                ColorHex("a52a2a"),
    "burlywood":            ColorHex("deb887"),
    "cadetblue":            ColorHex("5f9ea0"),
    "chartreuse":           ColorHex("7fff00"),
    "chocolate":            ColorHex("d2691e"),
    "coral":                ColorHex("ff7f50"),
    "cornflower":           ColorHex("6495ed"),
    "cornflowerblue":       ColorHex("6495ed"),
    "cornsilk":             ColorHex("fff8dc"),
    "crimson":              ColorHex("dc143c"),
    "cyan":                 ColorHex("00ffff"),
    "darkblue":             ColorHex("00008b"),
    "darkcyan":             ColorHex("008b8b"),
    "darkgoldenrod":        ColorHex("b8860b"),
    "darkgray":             ColorHex("a9a9a9"),
    "darkgreen":            ColorHex("006400"),
    "darkgrey":             ColorHex("a9a9a9"),
    "darkkhaki":            ColorHex("bdb76b"),
    "darkmagenta":          ColorHex("8b008b"),
    "darkolivegreen":       ColorHex("556b2f"),
    "darkorange":           ColorHex("ff8c00"),
    "darkorchid":           ColorHex("9932cc"),
    "darkred":              ColorHex("8b0000"),
    "darksalmon":           ColorHex("e9967a"),
    "darkseagreen":         ColorHex("8fbc8f"),
    "darkslateblue":        ColorHex("483d8b"),
    "darkslategray":        ColorHex("2f4f4f"),
    "darkslategrey":        ColorHex("2f4f4f"),
    "darkturquoise":        ColorHex("00ced1"),
    "darkviolet":           ColorHex("9400d3"),
    "deeppink":             ColorHex("ff1493"),
    "deepskyblue":          ColorHex("00bfff"),
    "dimgray":              ColorHex("696969"),
    "dimgrey":              ColorHex("696969"),
    "dodgerblue":           ColorHex("1e90ff"),
    "firebrick":            ColorHex("b22222"),
    "floralwhite":          ColorHex("fffaf0"),
    "forestgreen":          ColorHex("228b22"),
    "fuchsia":              ColorHex("ff00ff"),
    "gainsboro":            ColorHex("dcdcdc"),
    "ghostwhite":           ColorHex("f8f8ff"),
    "gold":                 ColorHex("ffd700"),
    "goldenrod":            ColorHex("daa520"),
    "gray":                 ColorHex("808080"),
    "green":                ColorHex("008000"),
    "greenyellow":          ColorHex("adff2f"),
    "grey":                 ColorHex("808080"),
    "honeydew":             ColorHex("f0fff0"),
    "hotpink":              ColorHex("ff69b4"),
    "indianred":            ColorHex("cd5c5c"),
    "indigo":               ColorHex("4b0082"),
    "ivory":                ColorHex("fffff0"),
    "khaki":                ColorHex("f0e68c"),
    "laserlemon":           ColorHex("ffff54"),
    "lavender":             ColorHex("e6e6fa"),
    "lavenderblush":        ColorHex("fff0f5"),
    "lawngreen":            ColorHex("7cfc00"),
    "lemonchiffon":         ColorHex("fffacd"),
    "lightblue":            ColorHex("add8e6"),
    "lightcoral":           ColorHex("f08080"),
    "lightcyan":            ColorHex("e0ffff"),
    "lightgoldenrod":       ColorHex("fafad2"),
    "lightgoldenrodyellow": ColorHex("fafad2"),
    "lightgray":            ColorHex("d3d3d3"),
    "lightgreen":           ColorHex("90ee90"),
    "lightgrey":            ColorHex("d3d3d3"),
    "lightpink":            ColorHex("ffb6c1"),
    "lightsalmon":          ColorHex("ffa07a"),
    "lightseagreen":        ColorHex("20b2aa"),
    "lightskyblue":         ColorHex("87cefa"),
    "lightslategray":       ColorHex("778899"),
    "lightslategrey":       ColorHex("778899"),
    "lightsteelblue":       ColorHex("b0c4de"),
    "lightyellow":          ColorHex("ffffe0"),
    "lime":                 ColorHex("00ff00"),
    "limegreen":            ColorHex("32cd32"),
    "linen":                ColorHex("faf0e6"),
    "magenta":              ColorHex("ff00ff"),
    "maroon":               ColorHex("800000"),
    "maroon2":              ColorHex("7f0000"),
    "maroon3":              ColorHex("b03060"),
    "mediumaquamarine":     ColorHex("66cdaa"),
    "mediumblue":           ColorHex("0000cd"),
    "mediumorchid":         ColorHex("ba55d3"),
    "mediumpurple":         ColorHex("9370db"),
    "mediumseagreen":       ColorHex("3cb371"),
    "mediumslateblue":      ColorHex("7b68ee"),
    "mediumspringgreen":    ColorHex("00fa9a"),
    "mediumturquoise":      ColorHex("48d1cc"),
    "mediumvioletred":      ColorHex("c71585"),
    "midnightblue":         ColorHex("191970"),
    "mintcream":            ColorHex("f5fffa"),
    "mistyrose":            ColorHex("ffe4e1"),
    "moccasin":             ColorHex("ffe4b5"),
    "navajowhite":          ColorHex("ffdead"),
    "navy":                 ColorHex("000080"),
    "oldlace":              ColorHex("fdf5e6"),
    "olive":                ColorHex("808000"),
    "olivedrab":            ColorHex("6b8e23"),
    "orange":               ColorHex("ffa500"),
    "orangered":            ColorHex("ff4500"),
    "orchid":               ColorHex("da70d6"),
    "palegoldenrod":        ColorHex("eee8aa"),
    "palegreen":            ColorHex("98fb98"),
    "paleturquoise":        ColorHex("afeeee"),
    "palevioletred":        ColorHex("db7093"),
    "papayawhip":           ColorHex("ffefd5"),
    "peachpuff":            ColorHex("ffdab9"),
    "peru":                 ColorHex("cd853f"),
    "pink":                 ColorHex("ffc0cb"),
    "plum":                 ColorHex("dda0dd"),
    "powderblue":           ColorHex("b0e0e6"),
    "purple":               ColorHex("800080"),
    "purple2":              ColorHex("7f007f"),
    "purple3":              ColorHex("a020f0"),
    "rebeccapurple":        ColorHex("663399"),
    "red":                  ColorHex("ff0000"),
    "rosybrown":            ColorHex("bc8f8f"),
    "royalblue":            ColorHex("4169e1"),
    "saddlebrown":          ColorHex("8b4513"),
    "salmon":               ColorHex("fa8072"),
    "sandybrown":           ColorHex("f4a460"),
    "seagreen":             ColorHex("2e8b57"),
    "seashell":             ColorHex("fff5ee"),
    "sienna":               ColorHex("a0522d"),
    "silver":               ColorHex("c0c0c0"),
    "skyblue":              ColorHex("87ceeb"),
    "slateblue":            ColorHex("6a5acd"),
    "slategray":            ColorHex("708090"),
    "slategrey":            ColorHex("708090"),
    "snow":                 ColorHex("fffafa"),
    "springgreen":          ColorHex("00ff7f"),
    "steelblue":            ColorHex("4682b4"),
    "tan":                  ColorHex("d2b48c"),
    "teal":                 ColorHex("008080"),
    "thistle":              ColorHex("d8bfd8"),
    "tomato":               ColorHex("ff6347"),
    "turquoise":            ColorHex("40e0d0"),
    "violet":               ColorHex("ee82ee"),
    "wheat":                ColorHex("f5deb3"),
    "white":                ColorHex("ffffff"),
    "whitesmoke":           ColorHex("f5f5f5"),
    "yellow":               ColorHex("ffff00"),
    "yellowgreen":          ColorHex("9acd32"),
}
